xcode/Subconscious/Shared/Components/Notebook/Notebook.swift

```diff
@@ -82,7 +82,12 @@ struct NotebookView: View {
         }
         /// Replay some app actions on notebook store
         .onReceive(
-            app.actions.compactMap(NotebookAction.from),
+            app.actions
+                .compactMap(NotebookAction.from)
+                .filter { action in
+                    action == .ready
+                        || app.state.selectedAppTab == .notebook
+                },
```

Comment:
Since this logic is beginning to become complex, we could consider factoring it out into a method of the view, or factoring out this filtered version of app actions out into a property of the view.

```swift
private var replayableAppActions: AnyPublisher<NotebookAction, Never> {
    app.actions
        .compactMap(NotebookAction.from)
        .filter {
            action == .ready
                || app.state.selectedAppTab == .notebook
        }
}
```

Reply:
Good idea, done.